### PR TITLE
Add error codes to CLI

### LINKED ISSUE #211: Define error scenarios and behaviors for CLI
If an error occurs while the tool runs, the CLI should return unsuccessful error codes.
We need to add this to the documentation.

### PR DESCRIPTION
**Pull request by anaismiller**
## Description

If an error occurs while the tool runs, the CLI should return unsuccessful error codes.
Closes #211 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request is clear and informative.
- [x] I have added myself to the 'assignees'.
- [x] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by VeraBE** [comment]
This PR could close #211

There are some "standards" on program exit codes, did you consider some of those?

File: README.md [hunk 1]
```diff
@@ -61,7 +61,18 @@ File: C:\Templates\azuredeploy.json
 #### SARIF
 Results are written to the file specified (with the `-o` or `--output-file-path` argument) in [SARIF](https://sarifweb.azurewebsites.net) format.
 
-### Understanding and Customizing Rules
+#### Error codes
+| Scenario      | Exit Code |
+| ----------- | ----------- |
+| Success: Operation is successful | 0 |
+| Error: GenericError | 1 |
```
Comment (VeraBE) [inline README.md:68]: Maybe it's worth reordering the list such that this one is at the end, and expressing that it corresponds to any error not previously mentioned?

File: README.md [hunk 2]
```diff
@@ -61,7 +61,18 @@ File: C:\Templates\azuredeploy.json
 #### SARIF
 Results are written to the file specified (with the `-o` or `--output-file-path` argument) in [SARIF](https://sarifweb.azurewebsites.net) format.
 
-### Understanding and Customizing Rules
+#### Error codes
```
Comment (VeraBE) [inline README.md:64]:
```suggestion
#### Exit codes
```

**Review by VeraBE — APPROVED** [review]

**Review by nonik0 — APPROVED** [review]

File: src/Analyzer.Cli/CommandLineParser.cs [hunk 3]
```diff
@@ -154,12 +154,12 @@ private int AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePa
...
                 writer.WriteResults(evaluations, (FileInfoBase)templateFilePath, (FileInfoBase)parametersFilePath);
 
-                return 1;
+                return 0;
             }
             catch (Exception exp)
             {
                 Console.WriteLine(GetExceptionMessage(exp));
-                return -1;
+                return 1;
```
Comment (JohnathonMohr) [inline src/Analyzer.Cli/CommandLineParser.cs:162]: @anaismiller This needs to be fixed. The returns here are for counting successes/failures when doing a directory scan. The CLI will not output correct counts now.

**Review by JohnathonMohr — COMMENTED** [review]
We need to make some changes to this.

**Pull request merged**